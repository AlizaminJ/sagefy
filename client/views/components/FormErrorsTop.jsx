import React from 'react'
import { shape } from 'prop-types'
import Icon from './Icon'

export default function FormErrorsTop({ formErrors }) {
  if (!Object.keys(formErrors).length) return null
  return (
    <div className="FormErrorsTop">
      <p>
        <mark>
          <Icon i="error" /> I couldn&apos;t do that...
        </mark>
      </p>
    </div>
  )
}

FormErrorsTop.propTypes = {
  formErrors: shape({}).isRequired,
}
