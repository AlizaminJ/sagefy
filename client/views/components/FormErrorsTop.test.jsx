import FormErrorsTop from './FormErrorsTop'

describe('FormErrorsTop component', () => {
  it('should render a FormErrorsTop null', () => {
    expect(FormErrorsTop({ formErrors: {} })).toMatchSnapshot()
  })

  it('should render a FormErrorsTop data', () => {
    expect(
      FormErrorsTop({
        formErrors: {
          all: [{ message: 'Hello' }],
        },
      })
    ).toMatchSnapshot()
  })
})
