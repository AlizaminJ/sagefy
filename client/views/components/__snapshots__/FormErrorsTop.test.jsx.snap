// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrorsTop component should render a FormErrorsTop data 1`] = `
<div
  className="FormErrorsTop"
>
  <p>
    <mark>
      <Icon
        i="error"
        s="m"
      />
       I couldn't do that...
    </mark>
  </p>
</div>
`;

exports[`FormErrorsTop component should render a FormErrorsTop null 1`] = `null`;
